omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/PatientLocationFragmentController.java

```diff
@@ -0,0 +1,134 @@
+                           UiUtils ui,
+                           FragmentConfiguration config,
+                           FragmentModel model) throws Exception {
+
+        Object patientConfig = config.get("patient");
+        if (patientConfig == null ) {
+            patientConfig = config.get("patientId");
+        }
+        Patient patient = null;
+        if (patientConfig != null) {
+            if (patientConfig instanceof Patient) {
+                patient = (Patient) patientConfig;
+            }
+            else if (patientConfig instanceof PatientDomainWrapper) {
+                patient = ((PatientDomainWrapper) patientConfig).getPatient();
+            }
+            else if (patientConfig instanceof Integer) {
+                patient = patientService.getPatient((Integer)patientConfig);
+            }
+            else if (patientConfig instanceof String) {
+                patient = patientService.getPatientByUuid((String)patientConfig);
+            }
+        }
+        if (patient == null) {
+            throw new IllegalArgumentException("No patient found. Please pass patient into the configuration");
+        }
+        VisitDomainWrapper activeVisit = adtService.getActiveVisit(patient, uiSessionContext.getSessionLocation());
+        String patientStatus = "";
+        String queueName = "";
+        Location currentInpatientLocation = null;
+        if (activeVisit != null ) {
+            // #1 Check if patient was admitted
+            InpatientAdmissionSearchCriteria criteria = new InpatientAdmissionSearchCriteria();
+            criteria.setPatientIds(Arrays.asList(patient.getPatientId()));
+            criteria.setIncludeDischarged(false);
+            List<InpatientAdmission> inpatientAdmissions = adtService.getInpatientAdmissions(criteria);
+            InpatientRequest currentInpatientRequest = null;
+            if (inpatientAdmissions != null && !inpatientAdmissions.isEmpty()) {
+                for (InpatientAdmission inpatientAdmission : inpatientAdmissions) {
+                    currentInpatientLocation = inpatientAdmission.getCurrentInpatientLocation();
+                    currentInpatientRequest = inpatientAdmission.getCurrentInpatientRequest();
+                    if (currentInpatientRequest != null) {
+                        String dispositionLocation = currentInpatientRequest.getDispositionLocation().getName();
+                        if (currentInpatientRequest.getDispositionType() == DispositionType.ADMIT) {
+                            patientStatus = ui.message("pihcore.waiting.tobe.admitted") + " " + dispositionLocation;
+                        } else if (currentInpatientRequest.getDispositionType() == DispositionType.TRANSFER) {
+                            patientStatus = ui.message("pihcore.waiting.for.transfer") + " " + dispositionLocation;
+                        } else {
+                            patientStatus = currentInpatientRequest.getDisposition().getName().getName();
```

Comment:
We might want to just display "Admitted" here if we don't find a status we recognize?

Reply:
@mogoodrich , I think this could be DISCHARGE which it should not happen in our system yet. But we could support this later? Maybe, I should return DISCHARGE then?

Reply:
Probably fine to keep as-is and just keep an eye on what might show up in it...